extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeDashManifestCreator.java

```diff
@@ -32,84 +34,72 @@
 import java.util.Map;
 import java.util.Objects;
 
-import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.addClientInfoHeaders;
-import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getYoutubeAndroidAppUserAgent;
-import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.isAndroidStreamingUrl;
-import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.isWebStreamingUrl;
-import static org.schabi.newpipe.extractor.utils.Utils.EMPTY_STRING;
-import static org.schabi.newpipe.extractor.utils.Utils.isBlank;
-import static org.schabi.newpipe.extractor.utils.Utils.isNullOrEmpty;
+import javax.annotation.Nonnull;
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
 
 /**
  * Class to generate DASH manifests from YouTube OTF, progressive and ended/post-live-DVR streams.
- *
- * <p>
  * It relies on external classes from the {@link org.w3c.dom} and {@link javax.xml} packages.
- * </p>
  */
 public final class YoutubeDashManifestCreator {
 
     /**
-     * URL parameter of the first sequence for live, post-live-DVR and OTF streams.
+     * The redirect count limit that this class uses, which is the same limit as OkHttp.
      */
-    private static final String SQ_0 = "&sq=0";
+    private static final int MAXIMUM_REDIRECT_COUNT = 20;
 
     /**
-     * URL parameter of the first stream request made by official clients.
+     * Cache of DASH manifests generated for OTF streams.
      */
-    private static final String RN_0 = "&rn=0";
+    private static final ManifestCreatorCache<String, String> OTF_CACHE
+            = new ManifestCreatorCache<>();
```

Comment:
It doesn't fit on the line above, and checkstyle enforces operators not to be at the end of lines, but rather at the start. You might be confused because in the app ktlint behaves the opposite way (but checkstyle works the same way)